File: src/main/java/hiiragi283/material/api/capability/fluid/ModuleMachineFluidTank.java
package hiiragi283.material.api.capability.fluid;

import hiiragi283.material.network.HiiragiFluidMessage;
import hiiragi283.material.network.HiiragiNetworkManager;
import hiiragi283.material.tile.TileEntityModuleMachine;
import org.jetbrains.annotations.Nullable;

public class ModuleMachineFluidTank extends HiiragiFluidTank {

    private final int index;

    public ModuleMachineFluidTank(int index, Type ioType, @Nullable TileEntityModuleMachine tile) {
        super(64000, ioType, tile);
        this.index = index;
    }

    @Override
    protected void onContentsChanged() {
        super.onContentsChanged();
        if (tile != null) {
            HiiragiNetworkManager.INSTANCE.sendToAll(new HiiragiFluidMessage(tile.getPos(), index, getFluid()));
        }
    }

}

File: src/main/kotlin/hiiragi283/material/api/capability/fluid/ModuleMachineFluidTank.kt
package hiiragi283.material.api.capability.fluid

import hiiragi283.material.api.capability.IOControllable
import hiiragi283.material.network.HiiragiMessage
import hiiragi283.material.network.HiiragiNetworkWrapper
import hiiragi283.material.tile.TileEntityModuleMachine

class ModuleMachineFluidTank(
    val index: Int,
    tile: TileEntityModuleMachine,
    override var ioType: IOControllable.Type = IOControllable.Type.CATALYST
) : HiiragiFluidTank(16000, ioType) {

    init {
        this.tile = tile
    }

    fun setIOType(type: IOControllable.Type): ModuleMachineFluidTank = also { ioType = type }

    override fun onContentsChanged() {
        tile?.let { HiiragiNetworkWrapper.sendToAll(HiiragiMessage.Client(tile.pos, tile.updateTag)) }
    }

}